baybe/simulation/core.py

```diff
@@ -0,0 +1,225 @@
+"""Core simulation and backtesting functionality."""
+
+from __future__ import annotations
+
+import warnings
+from copy import deepcopy
+from functools import partial
+from typing import Callable, Literal, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from baybe.campaign import Campaign
+from baybe.exceptions import NotEnoughPointsLeftError, NothingToSimulateError
+from baybe.simulation.lookup import _look_up_target_values
```

Comment:
Wondering that there is no complaint by any of our linters that we import a "private" function here? Would have expected this.

Reply:
it could also be fair to rename `simulation.lookup` into `simulation.utils` (like done in other modules) and make it a public function

Reply:
I'm now taking over YOUR position and declare: no renaming of functions happing in this PR. I did all this stuff in my original PR and you guys rightfully complained. This this PR will ONLY move stuff to new submodules. All the rest (renaming, private/public, changing signatures, refactoring) will get their own atomic PRs 🙃 

Renaming `lookup` to `utils` is debatable, but since it really only contains lookup functionality (and there will probably some more lookup stuff from `baybe.utils.dataframe` migrated), I think it could be useful to keep it as is and use `simulation.utils` for non-lookup utilities that will surely emerge during the remaining refactorings. So unless you have a very strong opinion on this, I'd say let's keep it for now

Reply:
Totally fine with not renaming now. I was actually just wondering that there was no liner complaint :D

Reply:
OK, I'll let @Scienfitz comment or resolve (+ accept the PR if everything else is fine) and then merge